src/resource-metadata/ec2.js

```diff
@@ -0,0 +1,30 @@
+const pLimit = require('p-limit');
+const pThrottle = require('p-throttle');
+const { schemaUrl, extractArchitecture, intAttr, stringAttr, traverse } = require("./common.js")
+
+async function collectEc2Resources() {
+    // TODO
+}
+
+exports.collectEc2Resources = collectEc2Resources
+
+function parseArn(resourceArn) {
+    const arn = resourceArn.split(":");
+    const partition = arn[1];
+    const service = arn[2];
+    if (arn.length > 6) {
+        const resourceId = arn[6];
+        const resourceType = partition + ":" + service + ":" + arn[5];
+        return [resourceType, resourceId]
+    } else {
+        const resource = splitOnce(arn[5], "/");
+        const resourceId = resource[1];
+        const resourceType = partition + ":" + service + ":" + resource[0];
+        return [resourceType, resourceId]
+    }
+}
```

Comment:
idk what are the domain names for these parameters. but the difference can be just extracted to a separate function
```
function extracted(arn) {
    if (arn.length > 6) {
        return {resourceElement:arn[5], resourceId: arn[6]};
    } else {
        const resource = splitOnce(arn[5], "/");
        return {resourceElement:resource[0], resourceId: resource[1]};
    }
}

function parseArn(resourceArn) {
    const arn = resourceArn.split(":");
    const partition = arn[1];
    const service = arn[2];
    const {resourceElement, resourceId} = extracted(arn);
    const resourceType = `${partition}:${service}:${resourceElement}`;
    return [resourceType, resourceId]
}
```